### PR TITLE
Update documentation about the support of private attachments for GBM and Facebook Messenger. RD-18781

### PR DESCRIPTION
**Pull request by onima**
* Jira Ticket : https://jira.ringcentral.com/browse/RD-18781

### CODE REVIEW

**Comment by alexeevit** [comment]
I have the only concern about that: some of our sources support private attachments and some don't. Ideally, we should explicitly mention for each source weather it supports private attachments. AFAIK, only Source SDK doesn't support private attachments.

We say "attachment should be public" in the common list of fields and then we add "it supports / doesn't support private attachments" in lists of specific fields of each source. I see you added that line to some sources but some of them are left without any information.

**Comment by onima** [comment]
`I have the only concern about that: some of our sources support private attachments and some don't. Ideally, we should explicitly mention for each source weather it supports private attachments. AFAIK, only Source SDK doesn't support private attachments.`
Yes ok I see, maybe we could add a new section in Overview part. Maybe a board like Channel Capabilities to list for each source if private attachments is supported as you suggested. WDY? https://developers.ringcentral.com/engage/digital/guide/interactions/structured-messages 

`We say "attachment should be public" in the common list of fields and then we add "it supports / doesn't support private attachments" in lists of specific fields of each source. I see you added that line to some sources but some of them are left without any information.`

I think what we could do here is just say in Primary Parameters for attachment_id field => `Support public and private attachments`. Then we don't need to redefine it each time for each structured content. For exemple on this page -> https://developers.ringcentral.com/engage/digital/guide/interactions/structured-messages/rich-link 
WDY ? :)

**Comment by onima** [comment]
@alexeevit

**Comment by alexeevit** [comment]
> Yes ok I see, maybe we could add a new section in Overview part. Maybe a board like Channel Capabilities to list for each source if private attachments is supported as you suggested. WDY?

Nah, too much honour :laughing:

> I think what we could do here is just say in Primary Parameters for attachment_id field => Support public and private attachments. Then we don't need to redefine it each time for each structured content.

I see what you mean. One of the main aim of the "Primary Parameters" is to provide most common requirements, like recommended values that are supported by all sources. So we shouldn't override the common requirements by more strict requirements. It's fine to override them by less strict requirements, thus users are still fine with using common requirements but they're also free to use features of a source.

Therefore, I think the best solution would be to keep it "Should be public" in the "Primary Parameters" and just check that all sources that support private attachments has a mention about that in the "Properties Unique to this Channel".

What do you think?

**Comment by onima** [comment]
Ok it should be good now : 
- SDK structured messages documentation specify that only public attachments are supported.
- I have updated documentation for each type of structured messages. It now shows for each type of structured message, which source supports private attachments.

File: docs/interactions/structured-messages/apple-auth.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ curl -X POST "https://[YOUR DOMAIN].api.digital.ringcentral.com/1.0/contents"
 | **Structured Content Settings** | | |
 | **`structured_content.type`** | String | Type of the structured message. Must be "authenticate". |
 | **`structured_content.response_body`** | String | **Optional**. Field to be the body of the user’s response. If absent, default is the structured message body. |
-| **`structured_content.attachment_id`** | Object | **Optional**. Existing attachment id used to decorate the authenticate structured message with an image.<br>Should be jpg, jpeg or png.<br>Should be less than 5MB. |
+| **`structured_content.attachment_id`** | Object | **Optional**. Existing attachment id used to decorate the authenticate structured message with an image.<br>Supports private attachments<br>Should be jpg, jpeg or png.<br>Should be less than 5MB. |
```
Comment (alexeevit) [inline docs/interactions/structured-messages/apple-auth.md:38]: Could you add a point after "Supports private attachments" in all Apple-specific types?

**Comment by alexeevit** [comment]
Thanks! Now it's almost splendid.

I found we have a problem in the documentation for carousels :smiley_cat: 
- It doesn't have unique properties for Facebook Messenger
- All unique properties for other sources are just copy-pasted from the documentation for templates. So all field names aren't correct (e.g. `structured_content.title` in GBM should be `structured_content.items.title` and `structured_content.items.title` should be `structured_content.items.items.title`)

Could you fix that?

**Comment by onima** [comment]
I have updated docs based on your comments. Please have a look and let me know if you're happy with new changes :) . Thx

File: docs/interactions/structured-messages/carousel.md [hunk 2]
```diff
@@ -95,9 +95,18 @@ curl -X POST "https://[YOUR DOMAIN].api.digital.ringcentral.com/1.0/contents"
...
-A carousel structured content is converted to a **Generic Template** with multiple elements in Facebook Messenger.
+### Properties Unique to this Channel
+
+Primary parameters are used by default, however, some parameters are unique or overwritten by parameters specific to this example.
 
-The same Template <**Generic Template**> specificities apply to every item of the carousel.
+| API Property | Specificity |
+|-|-|
+| **`structured_content.items.attachment_id`** | Supports bmp, gif, jpg, jpeg, png formats. Supports private attachments. [Upload attachments](../../../basics/uploads) for you own custom images. |
+| **`structured_content.items.url_text`** | **Ignored** property. |
```
Comment (alexeevit) [inline docs/interactions/structured-messages/carousel.md:105]:
```suggestion
| **`structured_content.items.url_text`** | Ignored property. |
```

**Comment by alexeevit** [comment]
There's only a small change request left. LGTM otherwise.
Good job! Thanks for the efforts!

**Comment by onima** [comment]
No prob. Thanks a lot for your help. Good teamwork 👍

**Comment by onima** [comment]
I don't know if you have the `power ` to approve this xD before I merge it. I will squash commit locally on git

**Comment by CoCraig** [comment]
> I don't know if you have the `power ` to approve this xD before I merge it. I will squash commit locally on git

@onima I just took a look at the proposed changes and I don't see any issues. Thanks for fixing the field names.  I can merge this as we are getting ready to publish.

**Comment by CoCraig** [comment]
> BTW, do you have a schedule of publishing we can have an access to?

Publishing is generally the end of each month.  There's a lot of flexibility though so sometimes we push it or change it depending on what needs to be updated. The best approach is to focus on the end of the month and reach out to me if you need an exact date lookup.

**Comment by CoCraig** [comment]
Thanks for the updates and for fixing some of the guides I moved over!

**Pull request merged**